### PR TITLE
[ez] Provide a slightly better error message if process times out

### PR DESCRIPTION
**Pull request by clee2000**
Just a slightly clearer error message

### CODE REVIEW

**Review by malfet — APPROVED** [review]

**Review by huydhn — APPROVED** [review]
LGTM!  You might want to follow up with an update on the log classifier if needed.

File: torch/testing/_internal/common_utils.py [hunk 1]
```diff
@@ -844,7 +844,9 @@ def wait_for_process(p, timeout=None):
             return exit_status
         else:
             p.kill()
-        raise
+        # Provide more info about the timeout (specifically that it timed out
+        # after the keyboard interrupt as well)
+        raise RuntimeError(f"Subprocess failed to exit smoothly after timeout {timeout} expired") from timeout_exception
```
Comment (huydhn) [inline torch/testing/_internal/common_utils.py:849]: Does this make sense to figure out a way to be provide more specific here like printing the name of the last slowest test?  I usually need to scan the log above for such information.  On the other hand, this is already an improvement over the current message from pytest.

Comment (clee2000) [reply torch/testing/_internal/common_utils.py:849]: That would need to go in run_test probably, since this is a more general function

**Pull request closed without merging**